.github/pull_request_template.md

```diff
@@ -0,0 +1,37 @@
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://github.com/m3db/m3/blob/master/CONTRIBUTING.md and developer notes https://github.com/m3db/m3/blob/master/DEVELOPER.md
+2. Please label this pull request according to what type of issue you are addressing.
+3. Ensure you have added or ran the appropriate tests for your PR: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#testing-changes
+4. Follow the instructions for writing a changelog note: read more at https://github.com/m3db/m3/blob/master/DEVELOPER.md#adding-a-changelog
+-->
+
+**What this PR does / why we need it**:
+
```

Comment:
nit, but can we add this to each section? (like you did for the release-note and documentation-note)
\```release-note
\```

Reply:
Hm no, those are used to specify sub snippets that may be automatically added to the changelog at a later time.

The text of the description will not be automatically appended anywhere except git history.